### PR TITLE
feat(async-sdk-cpp): added callbacks for all public&private (without unit tests)

### PR DESCRIPTION
**Pull request by leannejdong**
1. async callbacks have been added to all public and private APIs
2. some copy-paste errors (mostly on return type of `getAsync` methods) are corrected.

### CODE REVIEW

**Review by dendisuhubdy — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -52,3 +52,4 @@ java/.metadata/.plugins/org.eclipse.jdt.core/assumedExternalFilesCache
 java/.metadata/.plugins/org.eclipse.jdt.core/externalFilesCache
 java/.metadata/.plugins/org.eclipse.jdt.core/nonChainingJarsCache
 java/.metadata/.plugins/org.eclipse.ui.intro/introstate
```
Comment (sdmg15) [inline .gitignore]: Please remove all those lines and just put `java/*` inside the gitignore

File: cpp/source/rest/private/AccountBalance.hpp [hunk 2]
```diff
@@ -12,6 +12,13 @@ namespace Bitwyre::Rest::Private {
       return "/private/account/spotbalance";
     }
 
+    using Callback = std::function<void(const AccountBalanceResponse&)>;
```
Comment (sdmg15) [inline cpp/source/rest/private/AccountBalance.hpp]: Please make sure to put the type declarations at the beginning

File: cpp/source/rest/public/OrderTypes.hpp [hunk 3]
```diff
@@ -6,13 +6,19 @@ using AsyncOrderTypesResponse = std::future<OrderTypesResponse>;
...
     [[nodiscard]] static auto uri() noexcept -> std::string {
       return "/public/ordertypes";
     }
 
+    using Callback = std::function<void(const OrderTypesResponse&)>;
+    template<typename Dispatcher = Dispatcher>
+    [[nodiscard]] static auto getAsync(Callback cb) noexcept -> void {
+      auto result = getAsync();
+      return cb(result.get());
+    }//
```
Comment (sdmg15) [inline cpp/source/rest/public/OrderTypes.hpp]: Please remove those `//`

**Review by sdmg15 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (leannejdong) [reply .gitignore]: correct as requested. (These come from the pull from another developer. I should review before pulling. thanks)

File: cpp/source/rest/public/OrderTypes.hpp (hunk 3 above)
Comment (leannejdong) [reply cpp/source/rest/public/OrderTypes.hpp]: done

File: cpp/source/rest/private/AccountBalance.hpp (hunk 2 above)
Comment (leannejdong) [reply cpp/source/rest/private/AccountBalance.hpp]: done for all APIs

**Review by dendisuhubdy — APPROVED** [review]

**Review by sdmg15 — APPROVED** [review]

**Pull request merged**